Orange/widgets/visualize/owpythagoreanforest.py

```diff
@@ -192,11 +175,14 @@ def zoom_changed(self):
         self.grid.reflow(width)
         self.grid.setPreferredWidth(width)
 
-    # MODEL CHANGED METHODS
+    @contextmanager
+    def _prevent_commit(self):
```

Comment:
I like this use of context managers. However, you should add try-finally to ensure that `__prevent_commit` is set to `False` in case of exception. (https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager)